runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -5056,26 +5077,24 @@ static TR::Register * generateMultianewArrayWithInlineAllocators(TR::Node *node,
       iComment("Init 2st dim mustBeZero field.");
       cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, secondDimLenReg, generateS390MemoryReference(temp2Reg, fej9->getOffsetOfDiscontiguousArraySizeField(), cg));
       iComment("Init 2st dim size field.");
+   if (TR::Compiler->om.compressObjectReferences())
+      {
+      // Clear padding in contiguous array header layout. +4 because size field is 4 bytes wide.
+      cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, secondDimLenReg, generateS390MemoryReference(temp2Reg, static_cast<int32_t>(fej9->getOffsetOfDiscontiguousArraySizeField()) + 4, cg));
```

Comment:
Same question as before with firstDimLenReg.

Reply:
Because we are dealing with 0 size array at this point, and we used `LGF` to load dimension size, `secondDimLenReg` will be 0 so I figured we could use that to clear `dataAddr` field.